backend/windmill-api/src/slack_approvals.rs

```diff
@@ -0,0 +1,717 @@
+
+    let captures = re.captures(&action_value).ok_or_else(|| {
+        tracing::error!("Resume URL does not match the pattern.");
+        Error::BadRequest("Invalid URL format.".to_string())
+    })?;
+
+    let (w_id, action, job_id, resume_id, secret, approver) = (
+        captures.name("w_id").map_or("", |m| m.as_str()),
+        captures.name("action").map_or("", |m| m.as_str()),
+        captures.name("job_id").map_or("", |m| m.as_str()),
+        captures.name("resume_id").map_or("", |m| m.as_str()),
+        captures.name("secret").map_or("", |m| m.as_str()),
+        captures.name("approver").map(|m| m.as_str().to_string()),
+    );
+
+    let approver = QueryApprover { approver: approver };
+
+    let state_values = parse_state_values(&payload);
+    let state_json = serde_json::to_value(state_values).unwrap_or_else(|_| serde_json::json!({}));
+
+    tracing::debug!("W ID: {}, Action: {}, Job ID: {}, Resume ID: {}, Secret: {}, Approver: {:?}, State JSON: {:?}", 
+        w_id, action, job_id, resume_id, secret, approver.approver, state_json);
+
+    let job_uuid = Uuid::from_str(job_id).map_err(|_| {
+        Error::BadRequest("Invalid job ID format.".to_string())
+    })?;
+
+    let resume_id_parsed = resume_id.parse::<u32>().map_err(|_| {
+        Error::BadRequest("Invalid resume ID format.".to_string())
+    })?;
+
+    let res = resume_suspended_job(
+        authed,
+        Extension(db),
+        Path((
+            w_id.to_string(),
+            job_uuid,
+            resume_id_parsed,
+            secret.to_string(),
+        )),
+        Query(approver),
+        QueryOrBody(Some(state_json)),
+    )
+    .await;
+
+    tracing::debug!("resume_suspended_job: {:#?}", res);
+
+    if let Some(url) = response_url {
+        let message = if action == "resume" {
+            "\n\n*Workflow has been resumed!*"
+        } else {
+            "\n\n*Workflow has been canceled!*"
+        };
+        let _ = post_slack_response(&url, message).await;
+    }
+
+    Ok(StatusCode::OK)
+}
+
+pub async fn request_slack_approval(
+    authed: ApiAuthed,
+    Extension(db): Extension<DB>,
+    Path((w_id, job_id, resume_id)): Path<(String, Uuid, u32)>,
+    Query(approver): Query<QueryApprover>,
+    Query(message): Query<QueryMessage>,
+) -> windmill_common::error::JsonResult<serde_json::Value> {
+    let res = get_resume_urls(
+        authed,
+        axum::Extension(db.clone()),
+        Path((w_id, job_id, resume_id)),
+        axum::extract::Query(approver),
+    )
+    .await;
+
+    let schema: Option<ResumeFormRow> = sqlx::query_as!(
+        ResumeFormRow,
+        "SELECT
+            module.value->'suspend'->'resume_form' AS resume_form,
+            (module.value->'suspend'->>'hide_cancel')::boolean AS hide_cancel
+        FROM
+            job
+        LEFT JOIN
+            queue ON job.id = queue.parent_job
+        LEFT JOIN
+            jsonb_array_elements(job.raw_flow->'modules') AS module
```

Comment:
This will not work anymore, instead you can do something like this:
```rust
let (job_kind, script_hash, flow_step_id, raw_flow) = sqlx::query!(
    "SELECT
        job_kind AS \"job_kind: JobKind\",
        script_hash AS \"script_hash: ScriptHash\",
        flow_step_id,
        raw_flow AS \"raw_flow: Json<Box<RawValue>>\"
    FROM queue WHERE id = $1 AND workspace_id = $2 LIMIT 1",
    job_id,
    &w_id
)
.fetch_optional(&db)
.await
.and_then(|opt| opt.ok_or_else(|| error::Error::InternalError("not found".into())))
.map(|r| (r.job_kind, r.script_hash, r.flow_step_id, r.raw_flow))?;

let flow_data = windmill_common::cache::job::fetch_flow(&db, job_kind, script_hash)
    .or_else(|_| windmill_common::cache::job::fetch_preview_flow(&db, &job_id, raw_flow))
    .await?;
let flow_value = flow_data.value()?;

let module_step = Step::from_i32_and_len(flow_step_id, flow_value.modules.len());
let current_module = match module_step {
    Step::Step(i) => flow_value.modules.get(i),
    _ => None,
};

let schema = current_module.map(|module| {
    ResumeFormRow {
      ...
    }
});
```